kie-server-parent/kie-server-services/kie-server-services-common/src/test/java/org/kie/server/services/impl/marshal/MarshallerHelperTest.java

```diff
@@ -0,0 +1,183 @@
+
+	@Test
+	public void testMarshallWithoutContainer() {
+		KieServerRegistry kieServerRegistryMock = Mockito.mock(KieServerRegistry.class);
+
+		MarshallerHelper helper = new MarshallerHelper(kieServerRegistryMock);
+
+		QueryFilterSpec queryFilterSpec = new QueryFilterSpecBuilder().get();
+
+		String marshalledQFS = helper.marshal(MarshallingFormat.JAXB.toString(), queryFilterSpec);
+
+		String expectedMarshalledQFS = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>" + "<query-filter-spec>"
+				+ "<order-asc>false</order-asc>" + "</query-filter-spec>";
+
+		assertThat(marshalledQFS, CompareMatcher.isIdenticalTo(expectedMarshalledQFS).ignoreWhitespace());
+
+	}
+
+	@Test
+	public void testMarshallWithoutContainerWithExtraClasses() {
+		KieServerRegistry kieServerRegistryMock = Mockito.mock(KieServerRegistry.class);
+
+		Set<Class<?>> extraClasses = new HashSet<>();
+		extraClasses.add(TestExtraClass.class);
+
+		Mockito.when(kieServerRegistryMock.getExtraClasses()).thenReturn(extraClasses);
+
+		MarshallerHelper helper = new MarshallerHelper(kieServerRegistryMock);
+
+		TestExtraClass extraClass = new TestExtraClass();
+		extraClass.setBla("hallo");
+
+		String marshalledQFS = helper.marshal(MarshallingFormat.JAXB.toString(), extraClass);
+
+		String expectedMarshalledTEC = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>" + "<test-extra-class>"
+				+ "<bla>hallo</bla>" + "</test-extra-class>";
+
+		assertThat(marshalledQFS, CompareMatcher.isIdenticalTo(expectedMarshalledTEC).ignoreWhitespace());
+	}
+
+	/**
+	 * Tests that MarshallerHelper can also be used when passing in a <code>null</code> KieServerRegistry.
+	 */
+	@Test
+	public void testMarshallWithNullRegistry() {
+		MarshallerHelper helper = new MarshallerHelper(null);
+
+		QueryFilterSpec queryFilterSpec = new QueryFilterSpecBuilder().get();
+
+		String marshalledQFS = helper.marshal(MarshallingFormat.JAXB.toString(), queryFilterSpec);
+
+		String expectedMarshalledQFS = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>" + "<query-filter-spec>"
+				+ "<order-asc>false</order-asc>" + "</query-filter-spec>";
+
+		assertThat(marshalledQFS, CompareMatcher.isIdenticalTo(expectedMarshalledQFS).ignoreWhitespace());
+	}
+
+	@Test
+	public void testUnmarshallWithoutContainer() {
+		KieServerRegistry kieServerRegistryMock = Mockito.mock(KieServerRegistry.class);
+
+		MarshallerHelper helper = new MarshallerHelper(kieServerRegistryMock);
+
+		QueryFilterSpec expectedQueryFilterSpec = new QueryFilterSpecBuilder().get();
+
+		String marshalledQFS = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>" + "<query-filter-spec>"
+				+ "<order-asc>false</order-asc>" + "</query-filter-spec>";
+
+		QueryFilterSpec unmarshalledQFS = helper.unmarshal(marshalledQFS, MarshallingFormat.JAXB.toString(), QueryFilterSpec.class);
+
+		// QueryFilterSpec does not implement equals method .....
+		// assertEquals(expectedQueryFilterSpec, unmarshalledQFS);
```

Comment:
The problem is that that QueryFilterSpec does not implement an "equals" method. So either we:
- implement an "equals" method on that class.
- implement the equals manually
- remove the assert

Atm, I don't really have a preference. The thing is that we're testing MarshallerHelper, not Marshaller. The check whether the marshalled and unmarshalled content is correct should actually be in a Marshaller test.

Anyway, let me know what you prefer and I'll change the impl.